src/Sulu/Bundle/CategoryBundle/Admin/CategoryAdmin.php

```diff
@@ -93,20 +93,25 @@ public function configureViews(ViewCollection $viewCollection): void
         }
 
         if ($this->securityChecker->hasPermission(self::SECURITY_CONTEXT, PermissionTypes::EDIT)) {
-            $viewCollection->add(
-                $this->viewBuilderFactory
-                    ->createListViewBuilder(static::LIST_VIEW, '/categories/:locale')
-                    ->setResourceKey('categories')
-                    ->setListKey('categories')
-                    ->setTitle('sulu_category.categories')
-                    ->addListAdapters(['tree_table'])
-                    ->addLocales($locales)
-                    ->setDefaultLocale($locales[0])
-                    ->setAddView(static::ADD_FORM_VIEW)
-                    ->setEditView(static::EDIT_FORM_VIEW)
-                    ->enableSearching()
-                    ->addToolbarActions($listToolbarActions)
-            );
+            $listViewBuilder = $this->viewBuilderFactory
+                ->createListViewBuilder(static::LIST_VIEW, '/categories/:locale')
+                ->setResourceKey('categories')
+                ->setListKey('categories')
+                ->setTitle('sulu_category.categories')
+                ->addListAdapters(['tree_table'])
+                ->addLocales($locales)
+                ->setDefaultLocale($locales[0])
+                ->setEditView(static::EDIT_FORM_VIEW)
+                ->enableSearching()
+                ->addToolbarActions($listToolbarActions);
+
+            // set add view of list-view only if user has add permission
+            // this will hide the add button of the tree_table adapter if the user has no add permission
```

Comment:
Isn't that clear from the code below? I think I would have not added this comment, as it doesn't add any value to me, and duplicates stuff :see_no_evil:

Reply:
I would at least remove the first sentence, if not both. But the second one is ok for me, if you insist on it :see_no_evil: I just got bitten by wrong comments too many times in the past, therefore I am very careful about adding them if they don't add value.

Reply:
you are right, the first line is quite obvious. i added the comment because we dont do this in any other `Admin` class. the reason for this is that we dont use the `tree_table` adapter anywhere else and therefore it is not needed..